docs/dictionary/function/processor.lcdoc

```diff
@@ -23,17 +23,15 @@ if the processor is "ARM" then ...
 
 Returns (enum):
 
--   x86: Windows systems (any architecture) and Mac OS X and Linux
-    systems with a 32 bit processor.
--   x86_64: Mac and Linux systems running a 64 bit processor
--   i386: iOS Simulator.
--   ARM: iOS and Android Devices and the Android simulator
-
+-   x86: 32 or 64 bit Windows systems and 32 bit macOS or Linux systems.
```

Comment:
So the docs are very wrong (and always have been, if I've got this right) for 'the processor'. And the only real-world use I can see for this is to differentiate on a 64-bit system between a 32-bit and a 64-bit engine.

Looking into the ide, the processor is used on openstack in the home stack script to generate a filename when checking for updates, but that seems like an arbitrary naming convention. Same thing in revIDECheckForUpdates in revidelibrary.8.livecodescript. I realize I'm the one who issued the PR to fix that, but that was to get the installer to work with the existing naming convention, not to change the names to something compatible.

It's also used in revEnvironmentGuessRepositoryPath() to define the build directory, and that just seems wrong.

The only really useful scenario I can find for the current use of the processor is the function revIDEBrowserWidgetUnavailable() in revidelibrary.8.livecodescript where it appears that the browser widget isn't available for a 32-bit linux engine, even if running on a 64-bit system.